### PR TITLE
Add unit tests to bundle package

### PR DESCRIPTION
**Pull request by guillaumerose**
Before doing a larger refactoring, add unit tests.

### CODE REVIEW

File: pkg/crc/machine/bundle/metadata.go [hunk 1]
```diff
@@ -19,39 +19,56 @@ import (
...
+type DiskImage struct {
+	Name     string `json:"name"`
+	Format   string `json:"format"`
+	Size     string `json:"size"`
+	Checksum string `json:"sha256sum"`
+}
+
+type DriverInfo struct {
+	Name string `json:"name"`
 }
```
Comment (cfergeau) [inline pkg/crc/machine/bundle/metadata.go:66]: This hunk would make more sense in a separate commit after the commit adding the unit test, as this is exactly the kind of change where we need the unit test to make sure we did not break anything.

File: pkg/crc/machine/bundle/repository.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
+package bundle
+
```
Comment (cfergeau) [inline pkg/crc/machine/bundle/repository.go:2]: Is this code movement only? are there expected  behaviour changes? Would be useful to know in the commit log :)

File: pkg/crc/machine/bundle/repository.go [hunk 3]
```diff
@@ -39,26 +39,47 @@ func (bundle *CrcBundleInfo) readBundleInfo() error {
...
-	if err := bundleInfo.installSymlinkOrCopy(); err != nil {
+	return &bundleInfo, nil
+}
+
+func (r *Repository) Use(bundleName string) (*CrcBundleInfo, error) {
+	bundleInfo, err := r.Get(bundleName)
+	if err != nil {
+		return nil, err
+	}
+	if err := bundleInfo.installSymlinkOrCopy(r.OcBinDir); err != nil {
```
Comment (cfergeau) [inline pkg/crc/machine/bundle/repository.go:65]: This is pre-existing, but this method name is missing a hint that it is about the `oc` binary.

File: pkg/crc/machine/bundle/repository.go [hunk 4]
```diff
@@ -89,11 +81,23 @@ func (bundle *CrcBundleInfo) installSymlinkOrCopy(ocBinDir string) error {
...
+func (r *Repository) Extract(path string) error {
+	_, err := extract.Uncompress(path, r.CacheDir, true)
+	return err
+}
+
+var defaultRepo = &Repository{
+	CacheDir: constants.MachineCacheDir,
+	OcBinDir: constants.CrcOcBinDir,
+}
+
```
Comment (cfergeau) [inline pkg/crc/machine/bundle/repository.go:93]: The introduction of `defaultRepo` could have been squashed in the previous commit rather than here.

File: pkg/crc/machine/bundle/repository.go [hunk 5]
```diff
@@ -89,11 +81,23 @@ func (bundle *CrcBundleInfo) installSymlinkOrCopy(ocBinDir string) error {
 	return os.Symlink(ocInBundle, ocInBinDir)
 }
 
-func Extract(sourcepath string) (*CrcBundleInfo, error) {
-	_, err := extract.Uncompress(sourcepath, constants.MachineCacheDir, true)
-	if err != nil {
+func (r *Repository) Extract(path string) error {
```
Comment (cfergeau) [inline pkg/crc/machine/bundle/repository.go:84]: Wondering if this would be better as `func (repo *Repository)`? I usually have a hard time with one or two letter variable names.

**Review by cfergeau — COMMENTED** [review]
A high level summary of what the end goal is with the new `Repository` type could be useful?

File: pkg/crc/machine/bundle/metadata.go (hunk 1 above)
Comment (guillaumerose) [reply pkg/crc/machine/bundle/metadata.go:66]: I tried but the test change completely because the struct also change.
We don't break anything here because in the unit test, I take the raw json, unmarshall, marshall and compare to the raw json given as input.

File: pkg/crc/machine/bundle/repository.go (hunk 3 above)
Comment (guillaumerose) [reply pkg/crc/machine/bundle/repository.go:65]: done

File: pkg/crc/machine/bundle/repository.go (hunk 2 above)
Comment (guillaumerose) [reply pkg/crc/machine/bundle/repository.go:2]: Only move, added comment.

File: pkg/crc/machine/bundle/repository.go (hunk 4 above)
Comment (guillaumerose) [reply pkg/crc/machine/bundle/repository.go:93]: It was only once in the previous commit, the commit was already large, no need for it ;)

File: pkg/crc/machine/bundle/metadata.go (hunk 1 above)
Comment (cfergeau) [reply pkg/crc/machine/bundle/metadata.go:66]: It depends where you got the initial raw json from.  If it's the sample one from snc, then it's good, if it's one which was generated by a previous run of the test case, it's less good :)

File: pkg/crc/machine/bundle/repository.go (hunk 5 above)
Comment (guillaumerose) [reply pkg/crc/machine/bundle/repository.go]: done also :)

File: pkg/crc/machine/bundle/metadata.go (hunk 1 above)
Comment (guillaumerose) [reply pkg/crc/machine/bundle/metadata.go:66]: Took it from my local running crc installation. So, from the snc release CI.

**Review by cfergeau — APPROVED** [review]
